netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1174,6 +1174,7 @@ net_ebpf_ext_sock_addr_register_providers()
         _net_ebpf_sock_addr_hook_provider_data[i].bpf_attach_type =
             (bpf_attach_type_t)_net_ebpf_extension_sock_addr_bpf_attach_types[i];
         _net_ebpf_sock_addr_hook_provider_data[i].link_type = BPF_LINK_TYPE_CGROUP;
+        _net_ebpf_sock_addr_hook_provider_data[i].capabilities.support_extension_data_v1 = true;
```

Comment:
question: how do we ensure forward compatibility in such a case? for example, if the extension claims is supports `extension_data_v1`, but it is deployed on a system with an older eBPF runtime, how do we prevent any crash, etc.